plugin/trino-hive/src/test/java/io/trino/plugin/hive/BaseHiveConnectorTest.java

```diff
@@ -2893,6 +2895,26 @@ public void testUnregisterPartitionWithNullArgument()
         assertQueryFails("CALL system.unregister_partition('web', 'page_views', ARRAY['country'], NULL)", ".*partition_values cannot be null.*");
     }
 
+    @Test
+    public void testUnregisterTable()
+            throws IOException
+    {
+        String tableName = "test_unregister_table";
+        assertUpdate("" +
+                "CREATE TABLE " + tableName + " (" +
+                "  dummy_col bigint" +
+                ")");
+
+        assertUpdate(format("INSERT INTO %s (dummy_col) VALUES (1), (2), (3)", tableName), 3);
+
+        String tableLocation = (String) computeScalar("SELECT DISTINCT regexp_replace(\"$path\", '/[^/]*$', '') FROM " + tableName);
+        java.nio.file.Path tableDir = Paths.get(tableLocation.split(":")[1]);
```

Comment:
no need for file level checks - create an external table with the location of the unregistered table and make sure it still have the same content.